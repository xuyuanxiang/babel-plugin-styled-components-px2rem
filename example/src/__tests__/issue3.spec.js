import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import styled from 'styled-components';

const div = document.createElement('div');

beforeEach(() => {
  document.body.appendChild(div);
});

afterEach(() => {
  ReactDOM.unmountComponentAtNode(div);
  document.body.removeChild(div);
});

it('test 1', () => {
  TestUtils.act(() => {
    ReactDOM.render(
      React.createElement(
        styled.div`
          & {
            position: relative;
          }

          & {
            footer {
              height: 80px;
              background: green;
            }

            header {
              height: 45px;
              background: yellow;
            }
          }
        `,
        { id: 'testId' },
      ),
      div,
    );
  });
  expect(document.documentElement.innerHTML).toMatchSnapshot();
});

it('test 2', () => {
  TestUtils.act(() => {
    ReactDOM.render(
      React.createElement(
        styled.div`
          & {
            position: relative;
          }

          & footer {
            height: 80px;
            background: green;
          }

          & header {
            height: 45px;
            background: yellow;
          }
        `,
        { id: 'testId' },
      ),
      div,
    );
  });
  expect(document.documentElement.innerHTML).toMatchSnapshot();
});

it('test 3', () => {
  TestUtils.act(() => {
    ReactDOM.render(
      React.createElement(
        styled.div`
          & footer {
            height: 80px;
            background: green;
          }

          & header {
            height: 45px;
            background: yellow;
          }
        `,
        { id: 'testId' },
      ),
      div,
    );
  });
  expect(document.documentElement.innerHTML).toMatchSnapshot();
});
