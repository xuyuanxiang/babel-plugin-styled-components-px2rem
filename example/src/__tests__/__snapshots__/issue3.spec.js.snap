// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1 1`] = `
"<head><style data-styled=\\"\\" data-styled-version=\\"4.4.1\\">
/* sc-component-id: sc-bdVaJa */
.fAlAGr{position:relative;} .fAlAGr footer{height:0.8rem;background:green;} .fAlAGr header{height:0.45rem;background:yellow;}</style></head><body><div><div id=\\"testId\\" class=\\"sc-bdVaJa fAlAGr\\"></div></div></body>"
`;

exports[`test 2 1`] = `
"<head><style data-styled=\\"\\" data-styled-version=\\"4.4.1\\">
/* sc-component-id: sc-bdVaJa */
.fAlAGr{position:relative;} .fAlAGr footer{height:0.8rem;background:green;} .fAlAGr header{height:0.45rem;background:yellow;}
/* sc-component-id: sc-bwzfXH */
.mVqHc{position:relative;} .mVqHc footer{height:0.8rem;background:green;} .mVqHc header{height:0.45rem;background:yellow;}</style></head><body><div><div id=\\"testId\\" class=\\"sc-bwzfXH mVqHc\\"></div></div></body>"
`;

exports[`test 3 1`] = `
"<head><style data-styled=\\"\\" data-styled-version=\\"4.4.1\\">
/* sc-component-id: sc-bdVaJa */
.fAlAGr{position:relative;} .fAlAGr footer{height:0.8rem;background:green;} .fAlAGr header{height:0.45rem;background:yellow;}
/* sc-component-id: sc-bwzfXH */
.mVqHc{position:relative;} .mVqHc footer{height:0.8rem;background:green;} .mVqHc header{height:0.45rem;background:yellow;}
/* sc-component-id: sc-htpNat */
.eDoltG footer{height:0.8rem;background:green;} .eDoltG header{height:0.45rem;background:yellow;}</style></head><body><div><div id=\\"testId\\" class=\\"sc-htpNat eDoltG\\"></div></div></body>"
`;
